pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -319,6 +324,11 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
+	if foundUnifiedpushDeployment.Status.ReadyReplicas > 0 && instance.Status.UnifiedpushReady != true {
+		instance.Status.UnifiedpushReady = true
+		r.client.Status().Update(context.TODO(), instance)
+	}
```

Comment:
As you can see, this block is pretty much the same as the other one on L187 above. I'm not too bothered about the code duplication, but there are a couple of things I'm not sure about:

- Should I re-fetch the contents of the UPS resource (`instance` here) every time before checking (e.g. `instance.Status.UnifiedpushReady != true`)? It's already fetched at the beginning of this function, but maybe should be fetched again?
- Should I leave all of this until the end of the function, so that it's only calling the `r.client.Status().Update(context.TODO(), instance)` once? I guess that's maybe not the best idea, since there are many conditional returns before the end, which might mean that these don't get set promptly (I guess they'd eventually get set once everything else is okay).

Reply:
> Should I re-fetch the contents of the UPS resource (instance here) every time before checking (e.g. instance.Status.UnifiedpushReady != true)? It's already fetched at the beginning of this function, but maybe should be fetched again?

No, I don't think so. 

> Should I leave all of this until the end of the function, so that it's only calling the r.client.Status().Update(context.TODO(), instance) once? I guess that's maybe not the best idea, since there are many conditional returns before the end, which might mean that these don't get set promptly (I guess they'd eventually get set once everything else is okay).

I don't think it is a hard requirement, especially if the status field is important for the reconcile function itself. If it's only done once at the end, then there is a chance it will never be executed if there is an unexpected error occurred somewhere. I think it is ok to update the instance once there are actually changes made to it, but not to call update if there is no changes made.